main.cpp

```diff
@@ -45,8 +46,48 @@ struct Wrapper
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+
+private:
+    Type val;
+};
+
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& p) : point(std::move(p))
+    {
+        std::cout << "Wrapper(" << typeid(point).name() << ")" << std::endl;
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << point.toString() << ")" << std::endl;
+    }
+
+private:
+    Point point;
+};
+
+template<typename T>
+void variadicHelper(T&& first)
+{
+    Wrapper wrapper(std::forward<T>(first));
+    wrapper.print();
+}
+
+template<typename T, typename ... Args>
+void variadicHelper(T&& first, Args&& ... args)
+{
+    Wrapper wrapper ( std::forward<T>(first) );
+    wrapper.print();
+    variadicHelper( std::forward<Args>(args) ...);
+}

```

Comment:
Project9.1 challenge 2:
There is another recursive solution.  it is possible for Args ... to be empty.
your task: remove your implementation for instruction 4) and implement this alternate solution.
implement it after your existing variadicHelper function.
hint: what does the recursive call actually look like when called with an empty Variadic Parameter list?